src/movies.js

```diff
@@ -1,16 +1,108 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+function getAllDirectors(movies) {
+    let answer = movies.map(function(movie) {
+        return movie.director   
+    });
+    return answer
+}
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(movies) {
+    
+    let result = movies.filter(function(movie) {
+        // Condition: Only drama movies & from Steven Spielberg
+        if (movies.director == "Steven Spielberg" && movies.genre.includes("Drama")) {
+            return movie
+        }
+    });   
+return result   
+}
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(movies) {
+    let answer = movies.reduce(function(acc, elem) {
+        return acc + elem.rate
+    }, 0)
+return answer / movies.length
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
+function dramaMoviesRate(movies) {
+    let drama = movies.genre.includes("Drama");
+    if (drama) {
+        let answer = movies.reduce(function(acc, elem) {
+            return acc + elem.rate
+        }, 0)
+        return answer / drama
+    }
+} 
+
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
+function orderByYear(movies) {
+    let newSortedArray = JSON.parse(JSON.stringify(movies))
+   
+    newSortedArray.sort(function(a, b) {
+        if (a.year > b.year) {
+            return 1;
+        } else if (a.year < b.year) {
+            return -1;
+        } else if (a.year = b.year) {
+            return newSortedArray.sort(function(a, b) {
+                if (a.title > b.title) {
```

Comment:
There is no need to do another sort inside the first sort. Try removing lines 53 (the sort) and 61 (the closing brackets of the sort) and it will work.